$OpenBSD: patch-shell_cc-panel-loader_c,v 1.7 2020/05/14 15:20:35 ajacoutot Exp $

No GUdev on OpenBSD (XXX add HAVE_GUDEV to meson_options.txt)

http://permalink.gmane.org/gmane.comp.gnome.desktop/51950
https://bugzilla.gnome.org/show_bug.cgi?id=770758#c3
https://bugzilla.gnome.org/show_bug.cgi?id=766329
https://bugzilla.gnome.org/show_bug.cgi?id=772736 (systemd --user)

Index: shell/cc-panel-loader.c
--- shell/cc-panel-loader.c.orig
+++ shell/cc-panel-loader.c
@@ -42,7 +42,9 @@ extern GType cc_default_apps_panel_get_type (void);
 extern GType cc_display_panel_get_type (void);
 extern GType cc_info_overview_panel_get_type (void);
 extern GType cc_keyboard_panel_get_type (void);
+#ifdef NOTYET
 extern GType cc_mouse_panel_get_type (void);
+#endif
 #ifdef BUILD_NETWORK
 extern GType cc_network_panel_get_type (void);
 extern GType cc_wifi_panel_get_type (void);
@@ -54,7 +56,9 @@ extern GType cc_printers_panel_get_type (void);
 extern GType cc_region_panel_get_type (void);
 extern GType cc_removable_media_panel_get_type (void);
 extern GType cc_search_panel_get_type (void);
+#ifdef BUILD_SHARING
 extern GType cc_sharing_panel_get_type (void);
+#endif
 extern GType cc_sound_panel_get_type (void);
 #ifdef BUILD_THUNDERBOLT
 extern GType cc_bolt_panel_get_type (void);
@@ -106,7 +110,9 @@ static CcPanelLoaderVtable default_panels[] =
   PANEL_TYPE("location",         cc_location_panel_get_type,             NULL),
   PANEL_TYPE("lock",             cc_lock_panel_get_type,                 NULL),
   PANEL_TYPE("microphone",       cc_microphone_panel_get_type,           NULL),
+#ifdef NOTYET
   PANEL_TYPE("mouse",            cc_mouse_panel_get_type,                NULL),
+#endif
 #ifdef BUILD_NETWORK
   PANEL_TYPE("network",          cc_network_panel_get_type,              NULL),
   PANEL_TYPE("wifi",             cc_wifi_panel_get_type,                 cc_wifi_panel_static_init_func),
@@ -118,7 +124,9 @@ static CcPanelLoaderVtable default_panels[] =
   PANEL_TYPE("region",           cc_region_panel_get_type,               NULL),
   PANEL_TYPE("removable-media",  cc_removable_media_panel_get_type,      NULL),
   PANEL_TYPE("search",           cc_search_panel_get_type,               NULL),
+#ifdef BUILD_SHARING
   PANEL_TYPE("sharing",          cc_sharing_panel_get_type,              NULL),
+#endif
   PANEL_TYPE("sound",            cc_sound_panel_get_type,                NULL),
 #ifdef BUILD_THUNDERBOLT
   PANEL_TYPE("thunderbolt",      cc_bolt_panel_get_type,                 NULL),
